addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/MediaControlPlayer.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lgwebos.internal;
+
+import org.eclipse.smarthome.core.library.types.NextPreviousType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.library.types.RewindFastforwardType;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.lgwebos.handler.LGWebOSHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.connectsdk.device.ConnectableDevice;
+import com.connectsdk.service.capability.MediaControl;
+import com.connectsdk.service.capability.MediaControl.PlayStateListener;
+import com.connectsdk.service.capability.PlaylistControl;
+
+/**
+ * Handles commands of a Player Item.
+ *
+ *
+ * @author Sebastian Prehn
+ * @since 2.2.0
+ */
+public class MediaControlPlayer extends BaseChannelHandler<PlayStateListener> {
+    private Logger logger = LoggerFactory.getLogger(MediaControlPlayer.class);
+
+    private MediaControl getMediaControl(final ConnectableDevice device) {
+        return device.getCapability(MediaControl.class);
+    }
+
+    private PlaylistControl getPlayListControl(final ConnectableDevice device) {
+        return device.getCapability(PlaylistControl.class);
+    }
+
+    @Override
+    public void onReceiveCommand(final ConnectableDevice d, String channelId, LGWebOSHandler handler, Command command) {
+        if (d == null) {
+            return;
+        }
+        if (command instanceof NextPreviousType) {
```

Comment:
Are these `instanceof` checks needed, what would you think of something like

```java
switch(command) {
   case NextPreviousType.NEXT:
       if (d.hasCapabilities(PlaylistControl.Next)) {
           getPlayListControl(d).next(createDefaultResponseListener());
      }
      break;

...
```

Or just 6 if statements with a single else which catches the remaining state?

Reply:
Compile error for switch(command). Cannot switch on type Command.
But I can go without the instanceof checks, that's true - Done

Reply:
Yes of course you can only switch on enum types and given that the type of variable is actually an interface which is implemented by different enum's this is not going to work, sorry.